code/game/turfs/simulated/wall_icon.dm

```diff
@@ -177,6 +179,9 @@
 		add_overlay(SSmaterials.wall_damage_overlays[Clamp(round(damage / integrity * DAMAGE_OVERLAY_COUNT) + 1, 1, DAMAGE_OVERLAY_COUNT)])
 
 /turf/simulated/wall/proc/can_join_with(var/turf/simulated/wall/W)
-	if(material && istype(W.material) && get_wall_icon() == W.get_wall_icon())
-		return 1
+	if(material && istype(W.material))
+		if(LAZYACCESS(material.wall_blend_icons, W.get_wall_icon()))
```

Comment:
`LAZY*` macros are strictly slower than non-lazy, and materials are not numerous enough for empty lists to be a memory concern. It would be preferable to just have it default to an empty list and avoid using LAZYACCESS.